### PR TITLE
Added compression parameters selection to CLI

### LINKED ISSUE #359: CLI should allow selection of compression parameters
## Description

From @ni4 #294:

> rnp supports decompression and by default compresses data, but we actually don't have a way to enable/disable compression or change the compression level yet

```sh
# off=0 ctb=c1 tag=1 hlen=3 plen=268 new-ctb
:pubkey enc packet: version 3, algo 1, keyid D78F04BE3AF65209
	data: [2048 bits]
# off=271 ctb=d2 tag=18 hlen=2 plen=71 new-ctb
:encrypted data packet:
	length: 71
	mdc_method: 2
# off=284 ctb=c8 tag=8 hlen=2 plen=36 new-ctb
:compressed packet: algo=2
# off=287 ctb=cb tag=11 hlen=2 plen=27 new-ctb
:literal data packet:
	mode b (62), created 1502000813, name="hello.txt",
	raw data: 12 bytes
```

rnp already implements ZIP, BZIP and ZLIB.

Relevant GPG options:
```
-z n
Set compression level to n. A value of 0 for n disables compression. Default is to use the default compression level of zlib (normally 6).

--compress-algo n
Use compress algorithm n. Default is 2 which is RFC1950 compression. You may use 1 to use the old zlib version (RFC1951) which is used by PGP. The default algorithm may give better results because the window size is not limited to 8K. If this is not used the OpenPGP behavior is used, i.e. the compression algorithm is selected from the preferences; note, that this can't be done if you do not encrypt the data.
```

### PR DESCRIPTION
**Pull request by ni4**
- Added --zlib, --zip and --bzip2, -z 0..9 parameters to the rnp commandline and implemented their handling. Since compression now is used only during encryption it works only in this case.
- Fixed hanging bug in bzip2 decompression
- Updated CLI tests with more compression/decompression cases
- (hopefully) Fixed possible random CLI tests crash in Travis
- fixes #359

### CODE REVIEW

**Comment by ni4** [comment]
Looks like I'm lucky enough to hit #400
And another check hanged during the detached signature validation

File: src/tests/cli_tests.py [hunk 1]
```diff
@@ -144,18 +152,44 @@ def raise_err(msg, log = None):
         print log
     raise NameError(msg)
```
Comment (dewyatt) [inline src/tests/cli_tests.py]: Just a nitpick but `NameError` isn't really an appropriate error to raise here. Best practice would be to add a small class that derives from `Exception` and takes that optional log parameter (and override `__str__` to print that out if provided).

Comment (ni4) [reply src/tests/cli_tests.py]: Thanks, @dewyatt . It was just a quick pick, will fix it to work in a better way at some point.

**Comment by ni4** [comment]
I fixed hang during the signature verification, however still have #400 in one of the checks. 
Also updated exception throwing according to @dewyatt 's suggestion.
It seems that it would be a good idea to show less output with --debug so more tests could be run.

**Comment by dewyatt** [comment]
> It seems that it would be a good idea to show less output with --debug so more tests could be run.

I don't think we're anywhere near the travis limits on this, and the debug output really does help so I would prefer we keep it. You may hit the limit on the web interface, but you can always download the raw log (click on "Raw Log").

Some other minor suggestions for later:

* You may be interested in using the `--with-colons` option for `gpg --list-keys`, very easy to parse in a robust manner.
* It would be nice if we added a `prefix=rnp-cli` (or similar) to the `mkdtemp` call. Otherwise, when debugging, we end up with a bunch of directory names like `/tmp/tmpw0nT6G/` which are pretty generic sounding (it feels safer to `rm -rf /tmp/rnp-*`, rather than `rm -rf /tmp/tmp*`).

**Comment by ronaldtse** [comment]
Agree with both suggestions from @dewyatt . The last one is super easy to change.

@ni4 look like we can we rebase this and merge? This will solve #359

**Comment by ni4** [comment]
@ronaldtse I fixed prefix for mkdtemp, and now we know why tests are failing (see PR #429 ). So now this PR can be safely rebased and merged. Or should I do this?

**Comment by ronaldtse** [comment]
Thanks for finding that issue out with the RSA keys. @ni4 it would be great if you could rebase and merge it 👍

**Comment by ni4** [comment]
@ronaldtse Done, except I forgot to squash, sorry.

**Comment by ronaldtse** [comment]
No problem at all @ni4, thanks!

**Pull request merged**